session/manager.go

```diff
@@ -47,7 +51,8 @@ type ConfigNegotiator interface {
 type ConfigProvider func(consumerKey json.RawMessage) (ServiceConfiguration, DestroyCallback, error)
 
 // DestroyCallback cleanups session
-type DestroyCallback func() error
+// TODO get rid of this Destroy Callback and dryRun once we will have a better session lifecycle management.
+type DestroyCallback func(dryRun bool) error
```

Comment:
I was imaging that service(transport at that moment) should return session instance at creation, not just config provider and destroy callback. This session instance could be stored by the session manager. And session could provide methods like `config` `stop` `isAlive` or anything else related to the exact session.

Reply:
I don't think that session creation details should go to transport. We have session manager for session handling details. The only problem here that there is no way to communicate session lifecycle (death event to be exact) to outside world. And session creation could be quite complicated thing, which I don't want to move to transport.

Reply:
The problem now is that the session manager never knows if session inactive, there should be a method to detect if session inactive, but we even don't have an object that has such a method.

Reply:
It will know in short time - upcoming payment mechanism periodically communicates with service consumer and kill all inactive sesssions (for which no payments were received)

Reply:
As I said - session manager SHOULD track session inactivity itself, and signal any interested parties that session is gone. My suggestion was - some kind of channel on session struct.

Reply:
This PR was created as a quick fix to temporary resolve a problem with clean up wireguard interfaces. If we can wait for the upcoming payment mechanism it's perfect, and we can close this PR.

Reply:
It still needs some changes in session struct itself, so this PR can first step towards it - payments will reuse it too.

Reply:
DestroyCallback replaced by stop channel